### ISSUE TITLE
Block Name Already Exists - Block Factory Error

### ISSUE THREAD

**Issue by yurikleb** [google/blockly#7721]
### Check for duplicates

- [X] I have searched for similar issues before opening a new one.

### Description

I'm trying to create a new set of blocks using the "Block Factory" and export them via the "Workspace Factory".
However, if I try to go back to the "Block Factory Tab" from the "Workspace Factory" tab all my blocks get the warning icon next to the name with an error "you can't make a block xxxxx in this tool because that name already exists".

So every time I want to make some changes to my blocks I need to rename all of them.

### Reproduction steps

1.  Create one or more blocks in the "Block Factory" Tab
2. Click "Save Block" Button for all the blocks created
3. Click the "Workspace Factory" tab
4. Return to the "Block Factory" tab
5. See a warning icon appear in the blocks in the library - a warning about "that name already exists" forcing you to rename all your blocks

### Stack trace

```shell
From the Console:
You can't make a block called block_test in this tool because that name already exists.
warnExistingBlock @ factory.js:193
```

### Screenshots

![Screenshot 2023-12-17 at 22 35 45](https://github.com/google/blockly/assets/12280733/48ce19ec-3c3d-4ebe-b6a6-ace78993edeb)
![Screenshot 2023-12-17 at 22 38 15](https://github.com/google/blockly/assets/12280733/d40950db-bc9a-4bb6-b45c-3bc995edff9c)

### Browsers

Chrome desktop, Firefox desktop

**Comment by Apro123** [comment]
Similar issue here: https://github.com/google/blockly/issues/6820

**Comment by gonfunko** [comment]
This no longer appears to repro, and I believe it was resolved by #8605.
